web/packages/teleterm/src/ui/ConnectMyComputer/connectMyComputerContext.tsx

```diff
@@ -205,9 +234,40 @@ export const ConnectMyComputerContextProvider: FC<{
     }
   }
 
+  const agentIsNotStarted =
+    currentAction.kind === 'observe-process' &&
+    currentAction.agentProcessState.status === 'not-started';
+
+  useEffect(() => {
+    if (isAgentConfiguredAttempt.status === '') {
+      checkIfAgentIsConfigured();
+    }
+
+    const isAgentConfigured =
+      isAgentConfiguredAttempt.status === 'success' &&
+      isAgentConfiguredAttempt.data;
+    const shouldAutoStartAgent =
+      isAgentConfigured &&
+      canUse &&
+      workspacesService.getConnectMyComputerAutoStart(props.rootClusterUri) &&
+      agentIsNotStarted;
+    if (shouldAutoStartAgent) {
+      downloadAndStartAgent();
```

Comment:
After discussing the default step (https://github.com/gravitational/teleport/pull/30758#discussion_r1300279865), I was worried that the current implementation of auto start is somewhat implicit. I thought that the agent is going to autostart only because NavigationMenu calls `useConnectMyComputer`, and NavigationMenu is included in the default tab component.

But that's not true, is it? It works because it's called from an effect in a context and that context is created for every workspace.

Reply:
Exactly, the effect is placed in the context and it is called in the context-provider level. We can remove all `useConnectMyComputer` and it will still work.